client/my-sites/domains/domain-management/list/utils.js

```diff
@@ -1,5 +1,6 @@
 import { type } from 'calypso/lib/domains/constants';
 
 export const filterOutWpcomDomains = ( domains ) => {
-	return domains.filter( ( domain ) => domain.type !== type.WPCOM );
+	// domain.type might be lowercase, but type.WPCOM is uppercase
+	return domains.filter( ( domain ) => domain.type.toUpperCase() !== type.WPCOM );
```

Comment:
This also fixes this screen not showing, now it is showing:
![add your domain cta](https://github.com/Automattic/wp-calypso/assets/6586048/fa5dd5e8-1249-40a8-afcd-88e689882669)